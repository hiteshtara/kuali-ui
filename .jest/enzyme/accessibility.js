/* Copyright Â© 2016 Kuali, Inc. - All Rights Reserved
 * You may use and modify this code under the terms of the Kuali, Inc.
 * Pre-Release License Agreement. You may not distribute it.
 *
 * You should have received a copy of the Kuali, Inc. Pre-Release License
 * Agreement with this file. If not, please write to license@kuali.co.
 */

import assert from 'assert'
import { findDOMNode } from 'react-dom'
import { mount } from 'enzyme'
import axeCore from 'axe-core'

export async function assertComponentIsAccessible (component, axeConfig = {}) {
  const container = document.createElement('div')
  document.body.appendChild(container)

  const wrapper = mount(component, { attachTo: container })
  const nodeUnderTest = findDOMNode(wrapper.instance())

  const { violations = [] } = await axeCore.run(nodeUnderTest, axeConfig)

  for (const violation of violations) {
    const { help } = violation

    for (const node of violation.nodes) {
      const { target } = node
      assert.ok(false, `${help} on element ${target}`)
    }
  }

  document.body.removeChild(container)
}
