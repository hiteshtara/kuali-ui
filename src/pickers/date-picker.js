/* Copyright Â© 2016 Kuali, Inc. - All Rights Reserved
 * You may use and modify this code under the terms of the Kuali, Inc.
 * Pre-Release License Agreement. You may not distribute it.
 *
 * You should have received a copy of the Kuali, Inc. Pre-Release License
 * Agreement with this file. If not, please write to license@kuali.co.
 */
import React, { PureComponent } from 'react'
import PropTypes from 'prop-types'
import classnames from 'classnames'

import { DatePicker as MDDatePicker } from 'react-md'

export default class DatePicker extends PureComponent {
  static defaultProps = {
    ...DatePicker.defaultProps
  }

  static propTypes = {
    ...DatePicker.propTypes,
    pickerClassName: PropTypes.string,
    pickerHeaderClassName: PropTypes.string
  }

  render () {
    const { pickerClassName, pickerHeaderClassName, ...rest } = this.props
    const pickerClasses = classnames('kuali-date-picker', pickerClassName)
    const pickerHeaderClasses = classnames(
      'kuali-date-picker',
      pickerHeaderClassName
    )

    return (
      <MDDatePicker
        pickerClassName={pickerClasses}
        pickerHeaderClassName={pickerHeaderClasses}
        {...rest}
      />
    )
  }
}
