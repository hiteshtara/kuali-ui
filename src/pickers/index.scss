/* Copyright Â© 2016 Kuali, Inc. - All Rights Reserved
 * You may use and modify this code under the terms of the Kuali, Inc.
 * Pre-Release License Agreement. You may not distribute it.
 *
 * You should have received a copy of the Kuali, Inc. Pre-Release License
 * Agreement with this file. If not, please write to license@kuali.co.
 */

@include react-md-pickers;

.md-picker-header {
  background-color: #E0E0E0;
}

.md-picker-text {
  color: #333;
}

.kuali-time-picker {
  .md-time-periods {
    .md-picker-text {
      opacity: 0.5;
    }

    .md-picker-text--active {
      opacity: 1;
    }
  }
}

.kuali-date-picker {
  .md-calendar-month {
    .md-calendar-date {
      width: calc(99% / 7);
    }
  }

  .md-calendar-header {
    .md-calendar-dows {
      .md-calendar-date {
        width: calc(99% / 7);
      }
    }
  }

  .md-picker-header {
    .md-picker-control:last-child {
      h4 {
        transition: color 0s;
      }

      &:hover h4,
      &:focus h4 {
        color: palette(glacier);
      }
    }

    .md-picker-control:first-child {
      @extend .kuali-button;
      @extend .kuali-button--size__small;
      @extend .kuali-button--raised;

      margin-bottom: 10px;

      &.md-picker-text--active {
        @extend .kuali-button--info;
      }

      &:hover, &:focus {
        background-color: #DFDFDF;
      }
    }
  }
}
