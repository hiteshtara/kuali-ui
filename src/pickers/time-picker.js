/* Copyright Â© 2016 Kuali, Inc. - All Rights Reserved
 * You may use and modify this code under the terms of the Kuali, Inc.
 * Pre-Release License Agreement. You may not distribute it.
 *
 * You should have received a copy of the Kuali, Inc. Pre-Release License
 * Agreement with this file. If not, please write to license@kuali.co.
 */
import React, { PureComponent } from 'react'
import PropTypes from 'prop-types'
import classnames from 'classnames'

import { TimePicker as MDTimePicker } from 'react-md'

export default class TimePicker extends PureComponent {
  static defaultProps = {
    ...TimePicker.defaultProps
  }

  static propTypes = {
    ...TimePicker.propTypes,
    pickerClassName: PropTypes.string
  }

  render () {
    const { pickerClassName, ...rest } = this.props
    const pickerClasses = classnames('kuali-time-picker', pickerClassName)

    return <MDTimePicker pickerClassName={pickerClasses} {...rest} />
  }
}
