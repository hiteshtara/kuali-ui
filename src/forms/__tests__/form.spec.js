/* Copyright Â© 2016 Kuali, Inc. - All Rights Reserved
 * You may use and modify this code under the terms of the Kuali, Inc.
 * Pre-Release License Agreement. You may not distribute it.
 *
 * You should have received a copy of the Kuali, Inc. Pre-Release License
 * Agreement with this file. If not, please write to license@kuali.co.
 */

import React from 'react'
import { shallow } from 'enzyme'

import Form, { FieldGroup } from '..'
import TextField from '../../text-fields'
import SelectField from '../../select-fields'
import { DatePicker } from '../../pickers'
import {
  SelectionControl,
  SelectionControlGroup
} from '../../selection-controls'

describe('Form Component', () => {
  it('should render grid layout with grid lines', () => {
    const wrapper = shallow(<Form showGrid>{formContent}</Form>)
    expect(wrapper).toMatchSnapshot()
  })

  it('should render grid layout without grid lines', () => {
    const wrapper = shallow(<Form>{formContent}</Form>)
    expect(wrapper).toMatchSnapshot()
  })
})

const formContent = (
  <div>
    <FieldGroup>
      <TextField id='name' label='Name' className='md-cell kuali-form--cell' />
      <DatePicker
        id='birthdate'
        label='Birthdate'
        className='md-cell kuali-form--cell'
        autoOk
        formatOptions={{
          year: 'numeric',
          month: 'short',
          day: 'numeric'
        }}
      />
      <SelectField
        id='state'
        label='State'
        menuItems={[
          { name: 'Colorado', abbreviation: 'CO' },
          { name: 'Florida', abbreviation: 'FL' },
          { name: 'Indiana', abbreviation: 'IN' },
          { name: 'Utah', abbreviation: 'UT' }
        ]}
        itemLabel='name'
        itemValue='abbreviation'
        className='md-cell kuali-form--cell'
      />
      <TextField
        id='error'
        label='Error'
        className='md-cell kuali-form--cell'
        error
        errorText='Something went very wrong'
      />
    </FieldGroup>
    <FieldGroup>
      <div className='kuali-form--cell'>
        <label className='kuali-form--cell---label' htmlFor='hobbies'>
          Hobbies
        </label>
        <SelectionControl
          id='reading'
          name='hobbies'
          label='Reading'
          type='checkbox'
          value='reading'
        />
        <SelectionControl
          id='running'
          name='hobbies'
          label='Running'
          type='checkbox'
          value='running'
        />
        <SelectionControl
          id='swimming'
          name='hobbies'
          label='Swimming'
          type='checkbox'
          value='swimming'
        />
      </div>
      <div className='kuali-form--cell'>
        <SelectionControlGroup
          id='maritalstatus'
          type='radio'
          name='maritalstatus'
          label='Marital Status'
          defaultValue='single'
          controls={[
            {
              label: 'Single',
              value: 'single'
            },
            {
              label: 'Married',
              value: 'married'
            }
          ]}
        />
      </div>
    </FieldGroup>
    <FieldGroup>
      <div className='kuali-form--cell'>
        <label className='kuali-form--cell---label' htmlFor='citizen'>
          U.S. Citizen
        </label>
        <SelectionControl
          id='citizen'
          type='switch'
          name='citizen'
          defaultChecked
        />
      </div>
    </FieldGroup>
    <FieldGroup>
      <TextField
        id='lifestory'
        label='Life Story'
        className='md-cell kuali-form--cell'
        rows={2}
      />
    </FieldGroup>
  </div>
)
