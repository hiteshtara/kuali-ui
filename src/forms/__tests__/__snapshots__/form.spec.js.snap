// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form Component should render grid layout with grid lines 1`] = `
<form
  className="kuali-form kuali-form--gridded"
  style={Object {}}
>
  <div
    className="kuali-form--fieldset"
  >
    <div>
      <FieldGroup>
        <TextField
          className="md-cell kuali-form--cell"
          fullWidth={true}
          id="name"
          label="Name"
          leftIconStateful={true}
          lineDirection="left"
          passwordIcon={
            <FontIcon
              iconClassName="material-icons"
            >
              remove_red_eye
            </FontIcon>
          }
          rightIconStateful={true}
          type="text"
        />
        <DatePicker
          autoOk={true}
          className="md-cell kuali-form--cell"
          formatOptions={
            Object {
              "day": "numeric",
              "month": "short",
              "year": "numeric",
            }
          }
          id="birthdate"
          label="Birthdate"
        />
        <SelectField
          anchor={
            Object {
              "x": "inner left",
              "y": "overlap",
            }
          }
          className="md-cell kuali-form--cell"
          defaultValue=""
          defaultVisible={false}
          dropdownIcon={
            <FontIcon
              iconClassName="material-icons"
            >
              arrow_drop_down
            </FontIcon>
          }
          fixedTo={[Window]}
          id="state"
          itemLabel="name"
          itemProps="getProps"
          itemValue="abbreviation"
          keyboardMatchingTimeout={1000}
          label="State"
          lineDirection="left"
          menuItems={
            Array [
              Object {
                "abbreviation": "CO",
                "name": "Colorado",
              },
              Object {
                "abbreviation": "FL",
                "name": "Florida",
              },
              Object {
                "abbreviation": "IN",
                "name": "Indiana",
              },
              Object {
                "abbreviation": "UT",
                "name": "Utah",
              },
            ]
          }
          position="tl"
          repositionOnResize={false}
          repositionOnScroll={true}
          transitionName="md-drop"
          transitionTime={300}
        />
        <TextField
          className="md-cell kuali-form--cell"
          error={true}
          errorText="Something went very wrong"
          fullWidth={true}
          id="error"
          label="Error"
          leftIconStateful={true}
          lineDirection="left"
          passwordIcon={
            <FontIcon
              iconClassName="material-icons"
            >
              remove_red_eye
            </FontIcon>
          }
          rightIconStateful={true}
          type="text"
        />
      </FieldGroup>
      <FieldGroup>
        <div
          className="kuali-form--cell"
        >
          <label
            className="kuali-form--cell---label"
            htmlFor="hobbies"
          >
            Hobbies
          </label>
          <SelectionControl
            checkedCheckboxIcon={
              <FontIcon
                iconClassName="material-icons"
              >
                check_box
              </FontIcon>
            }
            checkedRadioIcon={
              <FontIcon
                iconClassName="material-icons"
              >
                radio_button_checked
              </FontIcon>
            }
            id="reading"
            label="Reading"
            name="hobbies"
            type="checkbox"
            uncheckedCheckboxIcon={
              <FontIcon
                iconClassName="material-icons"
              >
                check_box_outline_blank
              </FontIcon>
            }
            uncheckedRadioIcon={
              <FontIcon
                iconClassName="material-icons"
              >
                radio_button_unchecked
              </FontIcon>
            }
            value="reading"
          />
          <SelectionControl
            checkedCheckboxIcon={
              <FontIcon
                iconClassName="material-icons"
              >
                check_box
              </FontIcon>
            }
            checkedRadioIcon={
              <FontIcon
                iconClassName="material-icons"
              >
                radio_button_checked
              </FontIcon>
            }
            id="running"
            label="Running"
            name="hobbies"
            type="checkbox"
            uncheckedCheckboxIcon={
              <FontIcon
                iconClassName="material-icons"
              >
                check_box_outline_blank
              </FontIcon>
            }
            uncheckedRadioIcon={
              <FontIcon
                iconClassName="material-icons"
              >
                radio_button_unchecked
              </FontIcon>
            }
            value="running"
          />
          <SelectionControl
            checkedCheckboxIcon={
              <FontIcon
                iconClassName="material-icons"
              >
                check_box
              </FontIcon>
            }
            checkedRadioIcon={
              <FontIcon
                iconClassName="material-icons"
              >
                radio_button_checked
              </FontIcon>
            }
            id="swimming"
            label="Swimming"
            name="hobbies"
            type="checkbox"
            uncheckedCheckboxIcon={
              <FontIcon
                iconClassName="material-icons"
              >
                check_box_outline_blank
              </FontIcon>
            }
            uncheckedRadioIcon={
              <FontIcon
                iconClassName="material-icons"
              >
                radio_button_unchecked
              </FontIcon>
            }
            value="swimming"
          />
        </div>
        <div
          className="kuali-form--cell"
        >
          <SelectionControlGroup
            component="fieldset"
            controlComponent={[Function]}
            controls={
              Array [
                Object {
                  "label": "Single",
                  "value": "single",
                },
                Object {
                  "label": "Married",
                  "value": "married",
                },
              ]
            }
            defaultValue="single"
            id="maritalstatus"
            label="Marital Status"
            labelClassName="md-subheading-1"
            labelComponent="legend"
            name="maritalstatus"
            type="radio"
          />
        </div>
      </FieldGroup>
      <FieldGroup>
        <div
          className="kuali-form--cell"
        >
          <label
            className="kuali-form--cell---label"
            htmlFor="citizen"
          >
            U.S. Citizen
          </label>
          <SelectionControl
            checkedCheckboxIcon={
              <FontIcon
                iconClassName="material-icons"
              >
                check_box
              </FontIcon>
            }
            checkedRadioIcon={
              <FontIcon
                iconClassName="material-icons"
              >
                radio_button_checked
              </FontIcon>
            }
            defaultChecked={true}
            id="citizen"
            name="citizen"
            type="switch"
            uncheckedCheckboxIcon={
              <FontIcon
                iconClassName="material-icons"
              >
                check_box_outline_blank
              </FontIcon>
            }
            uncheckedRadioIcon={
              <FontIcon
                iconClassName="material-icons"
              >
                radio_button_unchecked
              </FontIcon>
            }
          />
        </div>
      </FieldGroup>
      <FieldGroup>
        <TextField
          className="md-cell kuali-form--cell"
          fullWidth={true}
          id="lifestory"
          label="Life Story"
          leftIconStateful={true}
          lineDirection="left"
          passwordIcon={
            <FontIcon
              iconClassName="material-icons"
            >
              remove_red_eye
            </FontIcon>
          }
          rightIconStateful={true}
          rows={2}
          type="text"
        />
      </FieldGroup>
    </div>
  </div>
</form>
`;

exports[`Form Component should render grid layout without grid lines 1`] = `
<form
  className="kuali-form"
  style={Object {}}
>
  <div
    className="kuali-form--fieldset"
  >
    <div>
      <FieldGroup>
        <TextField
          className="md-cell kuali-form--cell"
          fullWidth={true}
          id="name"
          label="Name"
          leftIconStateful={true}
          lineDirection="left"
          passwordIcon={
            <FontIcon
              iconClassName="material-icons"
            >
              remove_red_eye
            </FontIcon>
          }
          rightIconStateful={true}
          type="text"
        />
        <DatePicker
          autoOk={true}
          className="md-cell kuali-form--cell"
          formatOptions={
            Object {
              "day": "numeric",
              "month": "short",
              "year": "numeric",
            }
          }
          id="birthdate"
          label="Birthdate"
        />
        <SelectField
          anchor={
            Object {
              "x": "inner left",
              "y": "overlap",
            }
          }
          className="md-cell kuali-form--cell"
          defaultValue=""
          defaultVisible={false}
          dropdownIcon={
            <FontIcon
              iconClassName="material-icons"
            >
              arrow_drop_down
            </FontIcon>
          }
          fixedTo={[Window]}
          id="state"
          itemLabel="name"
          itemProps="getProps"
          itemValue="abbreviation"
          keyboardMatchingTimeout={1000}
          label="State"
          lineDirection="left"
          menuItems={
            Array [
              Object {
                "abbreviation": "CO",
                "name": "Colorado",
              },
              Object {
                "abbreviation": "FL",
                "name": "Florida",
              },
              Object {
                "abbreviation": "IN",
                "name": "Indiana",
              },
              Object {
                "abbreviation": "UT",
                "name": "Utah",
              },
            ]
          }
          position="tl"
          repositionOnResize={false}
          repositionOnScroll={true}
          transitionName="md-drop"
          transitionTime={300}
        />
        <TextField
          className="md-cell kuali-form--cell"
          error={true}
          errorText="Something went very wrong"
          fullWidth={true}
          id="error"
          label="Error"
          leftIconStateful={true}
          lineDirection="left"
          passwordIcon={
            <FontIcon
              iconClassName="material-icons"
            >
              remove_red_eye
            </FontIcon>
          }
          rightIconStateful={true}
          type="text"
        />
      </FieldGroup>
      <FieldGroup>
        <div
          className="kuali-form--cell"
        >
          <label
            className="kuali-form--cell---label"
            htmlFor="hobbies"
          >
            Hobbies
          </label>
          <SelectionControl
            checkedCheckboxIcon={
              <FontIcon
                iconClassName="material-icons"
              >
                check_box
              </FontIcon>
            }
            checkedRadioIcon={
              <FontIcon
                iconClassName="material-icons"
              >
                radio_button_checked
              </FontIcon>
            }
            id="reading"
            label="Reading"
            name="hobbies"
            type="checkbox"
            uncheckedCheckboxIcon={
              <FontIcon
                iconClassName="material-icons"
              >
                check_box_outline_blank
              </FontIcon>
            }
            uncheckedRadioIcon={
              <FontIcon
                iconClassName="material-icons"
              >
                radio_button_unchecked
              </FontIcon>
            }
            value="reading"
          />
          <SelectionControl
            checkedCheckboxIcon={
              <FontIcon
                iconClassName="material-icons"
              >
                check_box
              </FontIcon>
            }
            checkedRadioIcon={
              <FontIcon
                iconClassName="material-icons"
              >
                radio_button_checked
              </FontIcon>
            }
            id="running"
            label="Running"
            name="hobbies"
            type="checkbox"
            uncheckedCheckboxIcon={
              <FontIcon
                iconClassName="material-icons"
              >
                check_box_outline_blank
              </FontIcon>
            }
            uncheckedRadioIcon={
              <FontIcon
                iconClassName="material-icons"
              >
                radio_button_unchecked
              </FontIcon>
            }
            value="running"
          />
          <SelectionControl
            checkedCheckboxIcon={
              <FontIcon
                iconClassName="material-icons"
              >
                check_box
              </FontIcon>
            }
            checkedRadioIcon={
              <FontIcon
                iconClassName="material-icons"
              >
                radio_button_checked
              </FontIcon>
            }
            id="swimming"
            label="Swimming"
            name="hobbies"
            type="checkbox"
            uncheckedCheckboxIcon={
              <FontIcon
                iconClassName="material-icons"
              >
                check_box_outline_blank
              </FontIcon>
            }
            uncheckedRadioIcon={
              <FontIcon
                iconClassName="material-icons"
              >
                radio_button_unchecked
              </FontIcon>
            }
            value="swimming"
          />
        </div>
        <div
          className="kuali-form--cell"
        >
          <SelectionControlGroup
            component="fieldset"
            controlComponent={[Function]}
            controls={
              Array [
                Object {
                  "label": "Single",
                  "value": "single",
                },
                Object {
                  "label": "Married",
                  "value": "married",
                },
              ]
            }
            defaultValue="single"
            id="maritalstatus"
            label="Marital Status"
            labelClassName="md-subheading-1"
            labelComponent="legend"
            name="maritalstatus"
            type="radio"
          />
        </div>
      </FieldGroup>
      <FieldGroup>
        <div
          className="kuali-form--cell"
        >
          <label
            className="kuali-form--cell---label"
            htmlFor="citizen"
          >
            U.S. Citizen
          </label>
          <SelectionControl
            checkedCheckboxIcon={
              <FontIcon
                iconClassName="material-icons"
              >
                check_box
              </FontIcon>
            }
            checkedRadioIcon={
              <FontIcon
                iconClassName="material-icons"
              >
                radio_button_checked
              </FontIcon>
            }
            defaultChecked={true}
            id="citizen"
            name="citizen"
            type="switch"
            uncheckedCheckboxIcon={
              <FontIcon
                iconClassName="material-icons"
              >
                check_box_outline_blank
              </FontIcon>
            }
            uncheckedRadioIcon={
              <FontIcon
                iconClassName="material-icons"
              >
                radio_button_unchecked
              </FontIcon>
            }
          />
        </div>
      </FieldGroup>
      <FieldGroup>
        <TextField
          className="md-cell kuali-form--cell"
          fullWidth={true}
          id="lifestory"
          label="Life Story"
          leftIconStateful={true}
          lineDirection="left"
          passwordIcon={
            <FontIcon
              iconClassName="material-icons"
            >
              remove_red_eye
            </FontIcon>
          }
          rightIconStateful={true}
          rows={2}
          type="text"
        />
      </FieldGroup>
    </div>
  </div>
</form>
`;
